### PR TITLE
[review] Xml generation: check conflicting elements

### PR DESCRIPTION
**Pull request by dawagner**
Conflicting elements are elements belonging to more than one domain. The behaviour in such a case is unspecified.

In the context of XML generation from user input, this denotes a mistake from the user. We now warn him and fail the generation.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/329%23issuecomment-169950312%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/329%23discussion_r49187748%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/329%23issuecomment-170004971%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/329%23issuecomment-170836830%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/329%23discussion_r49456730%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/329%23discussion_r49460027%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/329%23discussion_r49462946%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/329%23issuecomment-169950312%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20please%20review.%22%2C%20%22created_at%22%3A%20%222016-01-08T09%3A52%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%20%22%2C%20%22created_at%22%3A%20%222016-01-08T13%3A33%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%22%2C%20%22created_at%22%3A%20%222016-01-12T08%3A31%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20576f4445e7f7b8641644cafc75f623f0a858b811%20test/xml-generator/test.py%2046%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/329%23discussion_r49456730%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22line%202%20long%22%2C%20%22created_at%22%3A%20%222016-01-12T14%3A03%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22nope%3A%20it%20is%2098%20chars%20long.%22%2C%20%22created_at%22%3A%20%222016-01-12T14%3A31%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222016-01-12T14%3A53%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/xml-generator/test.py%3AL90-132%22%7D%2C%20%22Pull%2037f085be71caaca56d5bdd9325355dac80ba8367%20test/xml-generator/test.py%2011%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/329%23discussion_r49187748%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%20-%20deep%20copy%20%3F%5Cr%5Cn%20-%20Would%20it%20not%20be%20better%20to%20have%20nominal_pfconfig%20a%20NON%20static%20variable%20to%20avoid%20all%20those%20copy.%22%2C%20%22created_at%22%3A%20%222016-01-08T13%3A33%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/xml-generator/test.py%3AL115-136%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/329#issuecomment-169950312'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard please review.
- <a href='https://github.com/krocard'></a> :-1:
- <a href='https://github.com/tcahuzax'></a> :-1:
- [ ] <a href='#crh-comment-Pull 37f085be71caaca56d5bdd9325355dac80ba8367 test/xml-generator/test.py 11'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/329#discussion_r49187748'>File: test/xml-generator/test.py:L115-136</a></b>
- <a href='https://github.com/krocard'></a> - deep copy ?
- Would it not be better to have nominal_pfconfig a NON static variable to avoid all those copy.
- [x] <a href='#crh-comment-Pull 576f4445e7f7b8641644cafc75f623f0a858b811 test/xml-generator/test.py 46'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/329#discussion_r49456730'>File: test/xml-generator/test.py:L90-132</a></b>
- <a href='https://github.com/krocard'></a> line 2 long
- <a href='https://github.com/dawagner'></a> nope: it is 98 chars long.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/329?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/329?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/329'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/329'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
@krocard please review.

File: test/xml-generator/test.py [hunk 1]
```diff
@@ -115,5 +115,21 @@ def test_nonfatalError(self):
...
         self.assertTrue(tester.check(self.nominal_reference, expectedErrors=1))
 
+    def test_conflicting(self):
+        vector = TestVector(edds=[os.path.join(vector_dir, "conflicting.pfw")])
+
+        tester = Tester(self.nominal_pfconfig, vector)
+        self.assertTrue(tester.check(expectedErrors=1))
+
+    def test_fail_criteria(self):
+        pfconfig = copy.copy(self.nominal_pfconfig)
```
Comment (krocard) [inline test/xml-generator/test.py]: - deep copy ?
 - Would it not be better to have nominal_pfconfig a NON static variable to avoid all those copy.

**Comment by krocard** [comment]
:-1:

**Comment by tcahuzax** [comment]
:-1:

File: test/xml-generator/test.py [hunk 2]
```diff
@@ -90,27 +90,42 @@ def check(self, reference=None, expectedErrors=0):
...
         tester.check(self.nominal_reference, expectedErrors=1)
 
+    def test_conflicting(self):
+        vector = TestVector(edds=[os.path.join(vector_dir, "conflicting.pfw")])
+
+        tester = Tester(self.nominal_pfconfig, vector)
+        tester.check(expectedErrors=1)
+
+    def test_fail_criteria(self):
+        self.nominal_pfconfig.criteria = os.path.join(config_dir, "duplicate_criterion_value.txt")
```
Comment (krocard) [inline test/xml-generator/test.py]: line 2 long

Comment (dawagner) [inline test/xml-generator/test.py]: nope: it is 98 chars long.

Comment (krocard) [inline test/xml-generator/test.py]: :+1:

Comment (krocard) [inline test/xml-generator/test.py]: :+1:<a href='#crh-update-none'></a>

File: test/xml-generator/test.py (hunk 1 above)
Comment (krocard) [inline test/xml-generator/test.py]: :+1:<a href='#crh-update-complete'></a>

**Comment by krocard** [comment]
:+1:<a href='#crh-update-none'></a>

**Pull request merged**